lib/media/drm_engine.js

```diff
@@ -1207,32 +1297,42 @@ shaka.media.DrmEngine = class {
       if (!present) {
         this.activeSessions_.delete(session);
 
+        const severity = this.config_.persistentSessionOnlinePlayback ?
+            shaka.util.Error.Severity.RECOVERABLE :
+            shaka.util.Error.Severity.CRITICAL;
+
         this.onError_(new shaka.util.Error(
-            shaka.util.Error.Severity.CRITICAL,
+            severity,
             shaka.util.Error.Category.DRM,
             shaka.util.Error.Code.OFFLINE_SESSION_REMOVED));
-        return Promise.resolve();
-      }
 
-      // TODO: We should get a key status change event.  Remove once Chrome CDM
-      // is fixed.
-      metadata.loaded = true;
-      if (this.areAllSessionsLoaded_()) {
-        this.allSessionsLoaded_.resolve();
+        metadata.loaded = true;
```

Comment:
Do we really need to set `metadata.loaded` to `true` here?
All that value does is determine when `areAllSessionsLoaded_` returns `true`. So if we are manually setting this to true for each stream we call `loadOfflineSession_` on, and we call `createOrLoad` on each session at once inside `createOrLoad`, doesn't that mean that we will just immediately resolve `this.allSessionsLoaded_` when the last session begins to load? That seems like it defeats the purpose of waiting for the key status change event, if we're still resolving before the event fires.

Reply:
We only set `metadata.loaded` when the session is not present anymore (we will never receive a `keystatuschange` event), and send a `OFFLINE_SESSION_REMOVED` event.

If other sessions are present & correctly loaded, we wait for the `keystatuschange` to set the session as loaded.